prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/HL7Utils.kt

```diff
@@ -21,71 +20,71 @@ object HL7Utils : Logging {
     private const val defaultHl7EncodingChars = "^~\\&#"
 
     /**
-     * The supported HL7 output messages.
+     * Gets a new object for the given [hl7Class].
+     * @return a message object
      */
-    enum class SupportedMessages(val type: Class<*>) {
-        ORU_R01_2_5_1(ORU_R01::class.java);
+    internal fun getMessage(hl7Class: String): Message {
+        return try {
+            val message = Class.forName(hl7Class).getDeclaredConstructor().newInstance()
+            if (message is Message) {
+                getMessageTypeString(message) // Just check we can get the type string
+                message
+            } else throw IllegalArgumentException("$hl7Class is not a subclass of ca.uhn.hl7v2.model.Message.")
+        } catch (e: Exception) {
+            throw IllegalArgumentException("$hl7Class is not a class to use for the conversion.")
+        }
+    }
+
+    /**
+     * Gets the type string for the given [message].
+     * @return a list with the message code and trigger event, or an empty list if the type could not be determined
+     */
+    internal fun getMessageTypeString(message: Message): List<String> {
+        val typeParts = message.javaClass.simpleName.split("_")
+        return if (typeParts.size != 2)
+            throw IllegalArgumentException("${message.javaClass.simpleName} is not a class to use for the conversion.")
+        else typeParts
+    }
+
+    /**
+     * Checks is a specific HL7 message [hl7Class] is supported.
```

Comment:
SP: `Checks if`

Reply:
Otherwise looks good to me, conversion works locally on my machine

Reply:
If you reviewed you may want to approve as well